### PR TITLE
change feature 9 design

### PR DESCRIPTION
**Pull request by mustafaozger**

### CODE REVIEW

File: editor-components/feature/feature9/feature9.tsx [hunk 1]
```diff
@@ -1,192 +1,289 @@
-import * as React from "react";
+import React, { useState,useEffect, useRef } from "react";
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:1]: remove unused imports

File: editor-components/feature/feature9/feature9.tsx [hunk 2]
```diff
@@ -1,192 +1,289 @@
...
+      <img alt="" className={styles.image} src={vertical_item.image}></img>
+      <h4 className={styles.title}>{vertical_item.title}</h4>
+      <h5 className={styles.title_description}>{vertical_item.description}</h5>
+      <br /><br /><br /><br /><br />      <br /><br /><br /><br />
+
+    </div>
+  );
+};
+
+let count="e"
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:74]: why count is a letter?

File: editor-components/feature/feature9/feature9.tsx [hunk 3]
```diff
@@ -1,192 +1,289 @@
...
+  id: string;
+  horizentals: VerticalItem[];
+};
+
+type ObserverCardProps = {
+  vertical_item: VerticalItem;
+  onIntersect: (id: string,isVisible:boolean) => void;
+};
+
+const ObserverCard: React.FC<ObserverCardProps> = ({ vertical_item, onIntersect }) => {
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:26]: we do not use functional components. remove this.

File: editor-components/feature/feature9/feature9.tsx [hunk 4]
```diff
@@ -1,192 +1,289 @@
...
-              key: "link",
-              displayer: "Button Link",
-              value: "",
-            },
             {
               type: "image",
               key: "image",
               value:
-                "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/644a673cf72de2002caaa361?alt=media&timestamp=1682597705111",
+                "https://res.cloudinary.com/disqnlpbh/image/upload/v1672312426/icon_40px_light_bulb_d6acf2aba1.svg",
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:113]: use google storage links for props

File: editor-components/feature/feature9/feature9.tsx [hunk 5]
```diff
@@ -1,192 +1,289 @@
...
+              value:
+                "We create a digital version of the best ideas from previous phase and create a medium-fidelity prototype.",
+              displayer: "Description",
             },
             {
               type: "image",
               key: "image",
               value:
-                "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/644a673cf72de2002caaa360?alt=media&timestamp=1682597705111",
+                "https://res.cloudinary.com/disqnlpbh/image/upload/v1672312426/icon_40px_layers_03432568ec.svg",
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:169]: use google storage links for props

File: editor-components/feature/feature9/feature9.tsx [hunk 6]
```diff
@@ -1,192 +1,289 @@
...
               value:
-                "Are you dreaming of a winter getaway? Look no further than these top 5 destinations for the perfect winter escape. From hitting the slopes in the Rocky Mountains for everyone.",
+                "Our design process is about making simple but eye-catching experiences. The primary goal is to achieve the wow factor and set you apart with that premium look.",
               displayer: "Description",
             },
+            {
+              type: "image",
+              key: "image",
+              value:
+                "https://res.cloudinary.com/disqnlpbh/image/upload/v1672312426/icon_40px_start_burst_d46533a76e.svg",
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:197]: use google storage links for props

File: editor-components/feature/feature9/feature9.tsx [hunk 7]
```diff
@@ -1,192 +1,289 @@
...
+              value:
+                "By leveraging the latest technologies, we share your brand and products with the world while focusing on a perfect visual output through stable and high-performing code.",
+              displayer: "Description",
             },
             {
               type: "image",
               key: "image",
               value:
-                "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/644a673cf72de2002caaa35f?alt=media&timestamp=1682597705111",
+                "https://res.cloudinary.com/disqnlpbh/image/upload/v1672312426/icon_40px_code_f3f0f45b00.svg",
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:225]: use google storage links for props

File: editor-components/feature/feature9/feature9.tsx [hunk 8]
```diff
@@ -1,192 +1,289 @@
...
-                        {horizontals.buttonText}
-                      </span>
-                    </ComposerLink>
-                  </div>
-                </div>
-                </div>
-              )
-            )}
+            <div className={this.decorateCSS("left-content")}>
+              <h1 className={this.decorateCSS("left_content_title")}>
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:250]: use kebab-case for classnames

File: editor-components/feature/feature9/feature9.tsx [hunk 9]
```diff
@@ -1,192 +1,289 @@
...
+import React, { useState,useEffect, useRef } from "react";
 import ComposerLink from "../../../../custom-hooks/composer-base-components/Link/link";
 import { BaseFeature } from "../../EditorComponent";
 import styles from "./feature9.module.scss";
+import { colors } from "@mui/material";
+import { object, string } from "yup";
 
-type Horizontals = {
+
+type VerticalItem = {
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:9]: give this a meaningful name

File: editor-components/feature/feature9/feature9.tsx [hunk 10]
```diff
@@ -1,192 +1,289 @@
...
+  id: string;
   title: string;
   description: string;
-  buttonText: string;
   image: string;
-  link: string;
 };
+
+type Verticals = {
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:16]: give this a meaningful name

File: editor-components/feature/feature9/feature9.tsx [hunk 11]
```diff
@@ -1,192 +1,289 @@
...
+  id: string;
+  horizentals: VerticalItem[];
+};
+
+type ObserverCardProps = {
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:21]: remove this

File: editor-components/feature/feature9/feature9.tsx [hunk 12]
```diff
@@ -1,192 +1,289 @@
...
+                      counterElement.innerText = id;
+                      counterElement.classList.add("changed");
+                      setTimeout(() => {
+                        counterElement.classList.remove("changed");
+                      }, 3000);
+                    }
+                  }}
+                />
+                )
+              )}
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:280]: remove this

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
please handle changes

File: editor-components/feature/feature9/feature9.tsx (hunk 3 above)
Comment (mustafaozger) [reply editor-components/feature/feature9/feature9.tsx:26]: I need to use IntersectionObserver to build my structure

File: editor-components/feature/feature9/feature9.tsx [hunk 13]
```diff
@@ -1,192 +1,280 @@
...
+        <img
+          alt="description image" 
+          className={this.decorateCSS("image")}
+          src={featureItem.image}
+        />
+        <h4 className={this.decorateCSS("sub-title")}>{featureItem.title}</h4>
+        <h5 className={this.decorateCSS("sub-title-description")}>
+          {featureItem.description}
+        </h5>
+        <br /><br/><br/><br /><br/><br/>
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:250]: remove br elements

File: editor-components/feature/feature9/feature9.tsx [hunk 14]
```diff
@@ -1,192 +1,280 @@
...
+        className={this.decorateCSS("card-item")}>
+        <img
+          alt="description image" 
+          className={this.decorateCSS("image")}
+          src={featureItem.image}
+        />
+        <h4 className={this.decorateCSS("sub-title")}>{featureItem.title}</h4>
+        <h5 className={this.decorateCSS("sub-title-description")}>
+          {featureItem.description}
+        </h5>
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:249]: don't render if h5 is empty
[check here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/feature/feature9/feature9.tsx [hunk 15]
```diff
@@ -1,192 +1,280 @@
...
+      <div 
+        key={index} 
+        id={`card-${index}`} 
+        className={this.decorateCSS("card-item")}>
+        <img
+          alt="description image" 
+          className={this.decorateCSS("image")}
+          src={featureItem.image}
+        />
+        <h4 className={this.decorateCSS("sub-title")}>{featureItem.title}</h4>
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:246]: don't render h4 if title is empty
[check here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/feature/feature9/feature9.tsx [hunk 16]
```diff
@@ -1,192 +1,280 @@
...
-                  </div>
-                </div>
-                </div>
-              )
-            )}
+            <div className={this.decorateCSS("main-title-content")}>
+              <h1 className={this.decorateCSS("main-title-content-text")}>
+                {this.getPropValue("main-title")}
+              </h1>
+            </div>
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.tsx:265]: don't render this if prop is empty
[read here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
please handle the changes and fix the bug

**Comment by mustafaozger** [comment]
I made changes but the features are still not loaded when adding the component for the first time. I am working on this bug. Also, new requests have been made and I will complete them as well.

File: editor-components/feature/feature9/feature9.module.scss [hunk 17]
```diff
@@ -62,7 +62,7 @@
 .sub-title-content{
   width: 50%;
   padding-right: 20%;
-  color: var(   --text-color-white) ;//white;
+  color: var(   --text-color-white) ;
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.module.scss:65]: i guess this is not the correct color variables. maybe it should be var(--composer-font-color-primary).
please check kardelen's discord message in general channel.

File: editor-components/feature/feature9/feature9.module.scss [hunk 18]
```diff
@@ -72,29 +72,33 @@
...
+  color: var(  --text-color-white);
 }
 
 .sub-title {
   display: block;
   margin: 0;
   font-size: 32px;
   font-weight: bold;
-  color: var(   --text-color-white);//white
+  color: var(   --text-color-white);
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.module.scss:101]: i guess this is not the correct color variables. maybe it should be var(--composer-font-color-primary).
please check kardelen's discord message in general channel.

File: editor-components/feature/feature9/feature9.module.scss [hunk 19]
```diff
@@ -72,29 +72,33 @@
...
+  opacity: 1; 
+}
 .image {
   margin: auto;
   max-width: 60px;
   max-height: 60px;
   min-width: 40px;
   padding: 10px;
-  color: var(  --text-color-white);//white
+  color: var(  --text-color-white);
```
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.module.scss:93]: why an image has font color?

File: editor-components/feature/feature9/feature9.module.scss
Comment (dogusmiuw) [inline editor-components/feature/feature9/feature9.module.scss:1]: please check all the color variables. most of them are wrong. check kardelen's discord message in general channel.

**Review by dogusmiuw — CHANGES_REQUESTED** [review]

File: editor-components/feature/feature9/feature9.tsx [hunk 20]
```diff
@@ -1,188 +1,362 @@
...
+    const cardsLengthIsChanged = this.getComponentState("cardLength") != cardElements.length;
+
+    if(cardsLengthIsChanged) {
+     this.setupObserver();
+   }
+   
     return (
-      <div className={this.decorateCSS("container")}>
+      <div
+        // ref={this.setupObserver}
```
Comment (Ayazelar) [inline editor-components/feature/feature9/feature9.tsx:268]:
```suggestion
```

**Review by Ayazelar — APPROVED** [review]

**Pull request merged**